test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,276 @@
+		_, err := createExternalNodeCRD(data, nsReturned.Name, vm.NodeName, vm.IFName, vm.IP)
+		require.Nil(t, err, "Failed to create ExternalNode")
+		startAntreaAgent(t, data, vm)
+	}
+
+	t.Logf("=== TestVerification ===")
+	for _, vm := range vmList {
+		hash := sha1.New()
+		io.WriteString(hash, vm.IFName)
+		hashedIfName := hex.EncodeToString(hash.Sum(nil))
+		// Generate ExternalEntity name from ExternalNode
+		eeName := vm.NodeName + "-" + hashedIfName[:interfaceNameLength]
+		// Verify ExternalEntities are created
+		if err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (done bool, err error) {
+			t.Logf("Fetching ExternalEntity %s", eeName)
+			ee, err := data.crdClient.CrdV1alpha2().ExternalEntities(nsReturned.Name).Get(context.TODO(), eeName, metav1.GetOptions{})
+			if err != nil {
+				t.Logf("Could not get ExternalEntity %s by ExternalNode %s", eeName, vm.NodeName)
+				return false, err
+			} else {
+				t.Logf("Got ExternalEntity %+v", ee)
+				return true, nil
+			}
+		}); err != nil {
+			assert.NoError(t, err, "Failed to get ExternalEntity %s by ExternalNode %s", eeName, vm.NodeName)
+		}
+	}
+
+	for _, vm := range vmList {
+		// Verify ExternalEntity is processed by antrea-agent
+		if err := wait.PollImmediate(30*time.Second, 1*time.Minute, func() (done bool, err error) {
+			t.Logf("Verify host interface configuration, VmInfo %+v", vm)
+			exists, err := verifyInterfaceIsInOVS(t, data, vm)
+			return exists, err
+		}); err != nil {
+			assert.NoError(t, err, "Failed to verify host interface in OVS, vmInfo %+v", vm)
+			continue
+		}
+
+		var tempVm vmInfo
+		if vm.OS == "Windows" {
+			tempVm = getWindowsVMInfo(t, data, vm.NodeName)
+		} else {
+			tempVm = getVMInfo(t, data, vm.NodeName)
+		}
+		assert.Exactly(t, vm.IFName, tempVm.IFName, "Failed to verify uplink interface, Expected %s, Got %s", vm.IFName, tempVm.IFName)
+		assert.Exactly(t, vm.IP, tempVm.IP, "Failed to verify uplink IP, Expected %s, Got %s", vm.IP, tempVm.IP)
+	}
+
+	t.Logf("=== TestCleanUp ===")
+	for _, vm := range vmList {
+		t.Logf("Deleting ExternalNode: %+v", vm.NodeName)
+		err := data.crdClient.CrdV1alpha1().ExternalNodes(nsReturned.Name).Delete(context.TODO(), vm.NodeName, metav1.DeleteOptions{})
+		require.Nil(t, err, "Failed to delete ExternalNode")
+		hash := sha1.New()
+		io.WriteString(hash, vm.IFName)
+		hashedIfName := hex.EncodeToString(hash.Sum(nil))
+		eeName := vm.NodeName + "-" + hashedIfName[:interfaceNameLength]
+		// Wait for ExternalEntities to be deleted
+		if err = wait.PollImmediate(10*time.Second, 1*time.Minute, func() (done bool, err error) {
+			t.Logf("Fetching ExternalEntity %s", eeName)
+			ee, err := data.crdClient.CrdV1alpha2().ExternalEntities(nsReturned.Name).Get(context.TODO(), eeName, metav1.GetOptions{})
+			if err != nil {
+				t.Logf("Could not get ExternalEntity %s", eeName)
+				return true, nil
+			} else {
+				t.Logf("ExternalEntity %s is not deleted yet", ee.Name)
+				return false, err
```

Comment:
Here err is nil, is it what you expected? In this case, the error in 156 is actually because a timeout error.